eth2/beacon/types/block_headers.py

```diff
@@ -0,0 +1,73 @@
+from typing import (
+    TYPE_CHECKING,
+)
+
+from eth_typing import (
+    BLSSignature,
+    Hash32,
+)
+from eth_utils import (
+    encode_hex,
+)
+
+import ssz
+from ssz.sedes import (
+    bytes32,
+    bytes96,
+    uint64,
+)
+
+from eth2.beacon.constants import EMPTY_SIGNATURE
+from eth2.beacon.typing import (
+    Slot,
+)
+
+
+if TYPE_CHECKING:
+    from eth2.beacon.db.chain import BaseBeaconChainDB  # noqa: F401
+
+
+class BeaconBlockHeader(ssz.SignedSerializable):
+
+    fields = [
+        ('slot', uint64),
+        ('previous_block_root', bytes32),
+        ('state_root', bytes32),
+        ('block_body_root', bytes32),
+        ('signature', bytes96),
+    ]
+
+    def __init__(self,
+                 *,
+                 slot: Slot,
+                 previous_block_root: Hash32,
+                 state_root: Hash32,
+                 block_body_root: Hash32,
+                 signature: BLSSignature=EMPTY_SIGNATURE):
+        super().__init__(
+            slot=slot,
+            previous_block_root=previous_block_root,
+            state_root=state_root,
+            block_body_root=block_body_root,
+            signature=signature,
+        )
+
+    def __repr__(self) -> str:
+        return '<BlockHeader #{0} {1}>'.format(
+            self.slot,
+            encode_hex(self.signing_root)[2:10],
```

Comment:
```suggestion
            encode_hex(self.root)[2:10],
```

IMO it could be confusing if we use the signing root here because it's possible that two headers only differ by their signature.

Reply:
That's a good point. I think that's because we now use `signing_root` as:
 1. `previous_block_root`
2.  the key in `ChainDB`

We can output both `signing_root` and `root`.